### DISCUSSION TITLE
nginx-proxy not forwarding to port 80 when deployed to docker swarm

### DISCUSSION THREAD

**Discussion by Rensuka** [nginx-proxy/nginx-proxy#1811] [Q&A]
Hi All,

I am attempting to learn docker and setting up a home networking thing. In doing so I wanted to use the nginx-proxy so I wouldn't need to remember all the different port numbers/ip addresses to access my deployed containers. The nginx-proxy is on the docker swarm but it is bound by constraints to a specific node. My issue is: the proxy works well, but it's not forwarding the ports from containers to port 80. Honestly, I don't know if this is actually intended as I don't see it explicitly mentioned anywhere. I'll provide an example down below.

```
docker service create \
  --name=transmission \
  -e PUID=1000 \
  -e PGID=1000 \
  -e VIRTUAL_HOST=transmission.suki.home \
  -e VIRTUAL_PORT=9091 \
  -e TRANSMISSION_WEB_HOME=/combustion-release/ \
  -e USER=user \
  -e PASS=<redacted> \
  -p 9091:9091 \
  -p 51413:51413 \
  -p 51413:51413/udp \
  --mount type=bind,source=/docker/persist/transmission,target=/config \
  --mount type=bind,source=/media/nas,destination=/media/nas \
  lscr.io/linuxserver/transmission
```

My expectation is that I would be able to go to http://transmission.suki.home in my browser and see the webui from that container.
The reality is that only http://transmission.suki.home:9091 will resolve to anything. Which is what I was trying to avoid.

Here is the command I was using to get nginx-proxy on my swarm.

```
docker service create -d \
  -p 80:80 \
  --name nginx-proxy \
  --mount type=bind,source=/var/run/docker.sock,destination=/tmp/docker.sock \
  -t jwilder/nginx-proxy
```

Thanks!

**Comment by junderw** [comment]
Try changing `-p 9091:9091 -p 51413:51413 -p 51413:51413/udp`

To `--expose=9091 --expose=51413/tcp --expose=51413/udp`

**Reply by Rensuka** [reply]
Thanks for the quick reply! --expose is not a valid flag for `docker service create` unfortunately

**Comment by junderw** [comment]
Also make sure `transmission.suki.home` is pointing to the nginx-proxy and not your transmission container directly.

**Reply by Rensuka** [reply]
Yep it is, I made sure of that before I posted.

**Comment by buchdag** [comment]
Could you give the result of `docker exec nginx-proxy nginx -T` ?

I'm not super familiar with Docker Swarm so you might have to adapt this command.
